client/client_test.go

```diff
@@ -223,36 +222,49 @@ func TestClient_Fingerprint(t *testing.T) {
 	require.NotEqual("", node.Attributes["driver.mock_driver"])
 }
 
-func TestClient_HasNodeChanged(t *testing.T) {
+func TestClient_TriggerNodeUpdate(t *testing.T) {
+	driver.CheckForMockDriver(t)
 	t.Parallel()
-	c := testClient(t, nil)
-	defer c.Shutdown()
 
-	node := c.config.Node
-	attrHash, err := hashstructure.Hash(node.Attributes, nil)
-	if err != nil {
-		c.logger.Printf("[DEBUG] client: unable to calculate node attributes hash: %v", err)
-	}
-	// Calculate node meta map hash
-	metaHash, err := hashstructure.Hash(node.Meta, nil)
-	if err != nil {
-		c.logger.Printf("[DEBUG] client: unable to calculate node meta hash: %v", err)
-	}
-	if changed, _, _ := c.hasNodeChanged(attrHash, metaHash); changed {
-		t.Fatalf("Unexpected hash change.")
-	}
+	// these constants are only defined when nomad_test is enabled, so these fail
```

Comment:
I fixed that so the comment isn't accurate and you probably don't need the nolint